reference/api/indexes.md

```diff
@@ -90,62 +94,73 @@ An index is automatically created when adding [documents](/reference/api/documen
 
 <CodeSamples id='create_an_index_1' />
 
-#### Response: `201 created`
+#### Response: `202 Accepted`
 
 ```json
 {
-  "uid": "movies",
-  "primaryKey": "movie_id",
-  "createdAt": "2019-11-20T09:40:33.711476Z",
-  "updatedAt": "2019-11-20T09:40:33.711476Z"
+  "uid": 0,
+  "indexUid": "movies",
+  "status": "enqueued",
+  "type": "indexCreation",
+  "enqueuedAt": "2021-08-12T10:00:00.000000Z"
 }
 ```
 
+You can use the response's `uid` to track the status of your request.
+
 ## Update an index
 
 <RouteHighlighter method="PUT" route="/indexes/:index_uid"/>
 
-Update an [index](/learn/core_concepts/indexes.md).
+Update an [index's](/learn/core_concepts/indexes.md) [primary key](/learn/core_concepts/documents.md#primary-key).
+
+If a primary key wasn't explicitly chosen during index creation, you can use this route to configure it. If you are unsure whether an index's primary key has already been configured, you can use [the get index route](/reference/api/indexes.md#get-one-index) to verify it.
 
-#### Path variables
+::: note
+It is not possible to change an index's `uid`.
+:::
+
+This is an asynchronous task. [You can read more about asynchronous processes in our dedicated guide.](/learn/advanced/asynchronous_updates.md)
+
+### Path variables
 
 | Variable      | Description   |
 | ------------- | ------------- |
 | **index_uid** | The index UID |
 
-#### Body
+### Body
 
 | Variable       | Description                                                |
 | -------------- | ---------------------------------------------------------- |
 | **primaryKey** | The <clientGlossary word="primary key" /> of the documents |
 
-The `uid` of an index cannot be changed.
-The `primaryKey` can be added if it does not already exist (to know if it has been set, use [the get index route](/reference/api/indexes.md#get-one-index)).
-
-[There are many ways in MeiliSearch to set the primary key](/learn/core_concepts/documents.md#primary-key).
-
 ### Example
 
 <CodeSamples id='update_an_index_1' />
 
-#### Response: `200 Ok`
+#### Response: `202 Accepted`
 
 ```json
 {
-  "uid": "movie_review",
-  "primaryKey": "movie_review_id",
-  "createdAt": "2019-11-20T09:40:33.711324Z",
-  "updatedAt": "2019-11-20T10:16:42.761858Z"
+  "uid": 1,
+  "indexUid": "movie_review",
+  "status": "enqueued",
+  "type": "indexUpdate",
+  "enqueuedAt": "2021-08-12T10:00:00.000000Z"
 }
 ```
 
+You can use the response's `uid` to track the status of your request.
```

Comment:
It would be nice to give a link to the API reference to get the task and watch the status.

Reply:
Yeap. I'm until that PR is approved so I can point the link to the correct page/heading.